From 7c2da625f6f18f2874e46006e47275a896eb2c82 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Wed, 15 Apr 2015 14:10:01 +0800
Subject: [PATCH] MLK-10659: sii902x audio: Fix imx7D hdmi audio failed work

imx7D hdmi audio will failed work after video mode changed.
It is cause by imx7D sii902x hdmi audio should setting to SAI
mode but sii902x default set audio to spdif mode in sii902x_setup
function. The function is called by every video mode change.
Move sii902x default audio mode code into probe function to
resolve the issue.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 drivers/video/mxc/mxsfb_sii902x.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/video/mxc/mxsfb_sii902x.c b/drivers/video/mxc/mxsfb_sii902x.c
index beddaf8..eb1bfb3 100644
--- a/drivers/video/mxc/mxsfb_sii902x.c
+++ b/drivers/video/mxc/mxsfb_sii902x.c
@@ -163,6 +163,10 @@ static void sii902x_setup(struct fb_info *fbi)
 	i2c_smbus_write_byte_data(sii902x.client, 0x09, 0x00);
 	/* set output format to RGB */
 	i2c_smbus_write_byte_data(sii902x.client, 0x0A, 0x00);
+}
+
+static void sii902x_audio_setup(void)
+{
 	/* audio setup */
 	i2c_smbus_write_byte_data(sii902x.client, 0x25, 0x00);
 	i2c_smbus_write_byte_data(sii902x.client, 0x26, 0x40);
@@ -429,6 +433,9 @@ static int sii902x_probe(struct i2c_client *client,
 		return -ENODEV;
 	}
 
+	/* enable hmdi audio */
+	sii902x_audio_setup();
+
 	/* try to read edid */
 	ret = sii902x_read_edid(&edid_fbi);
 	if (ret < 0)
-- 
1.8.0

